plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeHistoryTable.java

```diff
@@ -0,0 +1,148 @@
+                ImmutableList.<ColumnMetadata>builder()
+                        .add(new ColumnMetadata("version", BIGINT))
+                        .add(new ColumnMetadata("timestamp", TIMESTAMP_TZ_MILLIS))
+                        .add(new ColumnMetadata("user_id", VARCHAR))
+                        .add(new ColumnMetadata("user_name", VARCHAR))
+                        .add(new ColumnMetadata("operation", VARCHAR))
+                        .add(new ColumnMetadata("operation_parameters", typeManager.getType(mapType(VARCHAR.getTypeSignature(), VARCHAR.getTypeSignature()))))
+                        .add(new ColumnMetadata("cluster_id", VARCHAR))
+                        .add(new ColumnMetadata("read_version", BIGINT))
+                        .add(new ColumnMetadata("isolation_level", VARCHAR))
+                        .add(new ColumnMetadata("is_blind_append", BOOLEAN))
+                        //TODO add support for operationMetrics, userMetadata, engineInfo
+                        .build());
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        if (commitInfoEntries.isEmpty()) {
+            return new FixedPageSource(ImmutableList.of());
+        }
+        return new FixedPageSource(buildPages(session));
+    }
+
+    private List<Page> buildPages(ConnectorSession session)
+    {
+        PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
+        TimeZoneKey timeZoneKey = session.getTimeZoneKey();
+
+        commitInfoEntries.forEach(commitInfoEntry -> {
+            pagesBuilder.beginRow();
+
+            pagesBuilder.appendBigint(commitInfoEntry.getVersion());
+            pagesBuilder.appendTimestampTzMillis(commitInfoEntry.getTimestamp(), timeZoneKey);
+            write(commitInfoEntry.getUserId(), pagesBuilder);
+            write(commitInfoEntry.getUserName(), pagesBuilder);
+            write(commitInfoEntry.getOperation(), pagesBuilder);
+            if (commitInfoEntry.getOperationParameters() == null) {
+                pagesBuilder.appendNull();
+            }
+            else {
+                pagesBuilder.appendVarcharVarcharMap(commitInfoEntry.getOperationParameters());
+            }
+            write(commitInfoEntry.getClusterId(), pagesBuilder);
+            pagesBuilder.appendBigint(commitInfoEntry.getReadVersion());
+            write(commitInfoEntry.getIsolationLevel(), pagesBuilder);
+            if (commitInfoEntry.isBlindAppend().isEmpty()) {
+                pagesBuilder.appendNull();
+            }
+            else {
+                write(commitInfoEntry.isBlindAppend().get(), pagesBuilder);
```

Comment:
nit: How about removing `write(Boolean value, PageListBuilder pagesBuilder)` method and inline `pagesBuilder.appendBoolean` ? The null check of `write` method seems redundant.